src/metrics/metrics.go

```diff
@@ -39,8 +39,51 @@ var MetricsRequestedGasLimit = prometheus.NewHistogramVec(
 	[]string{"encrypted_tx_hash"},
 )
 
+var MetricsUpstreamRequestDuration = prometheus.NewHistogramVec(
```

Comment:
You can remove the metrics prefix in all the metrics names, as it's redundant with the package also being named metrics.